### PR TITLE
add an argument to use only labeled data in training process.

### PR DESCRIPTION
**Pull request by DoctorKey**
Maybe, it is a simple change that can resolve the problem mentioned by #6 .

### CODE REVIEW

**Comment by avital** [comment]
This looks probably good, but one thing: this is changing the effective batch size for the fully supervised experiments to be twice as large as original code.

Because batch size can affect the results of experiments, I think we should explicitly make the batch size half the size in the .yml files.

What do you think?

**Comment by DoctorKey** [comment]
BN layers use 100 batch size in original code. Just making the batch size half will change the computation of BN layers, although it maintains the effective batch size for loss. Maybe, there is a need to optimize the hyperparameters again.

**Comment by avital** [comment]
Here's what I propose we do:

Run all of the fully-supervised baselines with this pull request, in two ways: with batch size 100 and batch size 50. Hopefully the results end up being very close to what we have reported in the paper.

In case batch size of 50 or 100 matters, or if the results end up much different than our original published results, then we may want to re-tune the hyperparameters, which will take longer.

What do you think?

**Comment by DoctorKey** [comment]
OK. Good luck to you.

File: lib/data_provider.py [hunk 1]
```diff
@@ -53,6 +54,9 @@ def get_simple_mixed_batch(
             key.  Returns a boolean tensor. Defaults to no filter (all images).
         unlabeled_data_filter_fn (function): Function to decide which unlabeled
             data to look at. Same signature as labeled_data_filter.
+        mode (str): "labeled" - use only labeled data,
+                    "unlabeled" - use only unlabel data,
```
Comment (avital) [inline lib/data_provider.py]:
```suggestion
```

(I don't know how to train a classifier with only unlabeled data)

File: train_model.py [hunk 2]
```diff
@@ -82,6 +82,13 @@
     "datasets being used as unlabeled data. Defaults to all "
     "classes.",
 )
+flags.DEFINE_string(
+    "dataset_mode",
+    "mix",
+    "'labeled' - use only labeled data to train the model. "
+    "'unlabeled' - use only unlabel data to train the model"
```
Comment (avital) [inline train_model.py]:
```suggestion
```

Comment (avital) [reply train_model.py]: (I don't know how to train a classifier with only unlabeled data)

File: lib/data_provider.py [hunk 3]
```diff
@@ -124,9 +128,11 @@ def get_simple_mixed_batch(
 
     # These operations merge the datasets in a way that intersperses
     # elements from each, rather than just concatenating them.
-    if labeled_dataset_name == "imagenet_32":
+    if labeled_dataset_name == "imagenet_32" or mode == "labeled":
         # Don't waste batch space for imagenet pre-training.
         dataset = labeled_dataset
+    elif mode == "unlabeled":
```
Comment (avital) [inline lib/data_provider.py:133]: Please also remove this clause here, per my other comments -- it doesn't make sense to train a classifier with only unlabeled data.

File: lib/data_provider.py [hunk 4]
```diff
@@ -124,9 +128,11 @@ def get_simple_mixed_batch(
...
+        dataset = unlabeled_dataset
     else:
         dataset = dataset_utils.shuffle_merge(
```
Comment (avital) [inline lib/data_provider.py:137]:
```suggestion
        assert mode == "mix"
        dataset = dataset_utils.shuffle_merge(
```

**Comment by avital** [comment]
Hi @DoctorKey, thanks again for the PR. I made some comments, please take a look. Also, have you run this code? Can you report the results for CIFAR-10 fully supervised and SVHN fully-supervised before and after this change? (the table-1 runs).

**Comment by JiechengZhao** [comment]
@DoctorKey Thanks for the code, we are also wondering the effect that unlabeled data acts on BN layer. We use your code to have a test on SVHN. there is merely no change in the result. But we only run parts of the experiment for the limited resources.

**Comment by DoctorKey** [comment]
I run the code when I made this PR. Because of the limited resources, I only got the result of CIFAR-10 fully supervised. We found that the result was almost unchanged.

**Comment by diego898** [comment]
hey @avital - just a friendly ping to see if this PR (and repo) are still planned to be merged. Thanks!

**Comment by avital** [comment]
Merged, thanks for the ping.
